From f79ace23917752a622a660dabec502582cacdef9 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 7 Jul 2020 10:55:16 -0700
Subject: [PATCH 1426/1851] v4l: xilinx: scd: Correct max/min supported
 resolutions

commit 92db145a176d230ceba19ab1179f378997b4b321 from
https://github.com/Xilinx/linux-xlnx.git

As per Xilinx SCD spec (pg322), SCD can support up to 8192x4320 and down
to 64x64. Currently the max/min resolutions are incorrect so values are
incorrectly clamped during S_FMT. This patch corrects the max/min
resolutions so that users can actually set these resolutions during
S_FMT.

Also to not change the default resolution/behavior, this patch keeps the
default format as 3840x2160.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../platform/xilinx/xilinx-scenechange-channel.c   | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
index 4f3a8d03d217..a3c841a9f3aa 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
@@ -19,10 +19,12 @@
 #include "xilinx-scenechange.h"
 #include "xilinx-vip.h"
 
-#define XSCD_MAX_WIDTH		3840
-#define XSCD_MAX_HEIGHT		2160
-#define XSCD_MIN_WIDTH		640
-#define XSCD_MIN_HEIGHT		480
+#define XSCD_DEFAULT_WIDTH	3840
+#define XSCD_DEFAULT_HEIGHT	2160
+#define XSCD_MAX_WIDTH		8192
+#define XSCD_MAX_HEIGHT		4320
+#define XSCD_MIN_WIDTH		64
+#define XSCD_MIN_HEIGHT		64
 
 #define XSCD_V_SUBSAMPLING		16
 #define XSCD_BYTE_ALIGN			16
@@ -369,8 +371,8 @@ int xscd_chan_init(struct xscd_device *xscd, unsigned int chan_id,
 	/* Initialize default format */
 	chan->format.code = MEDIA_BUS_FMT_VYYUYY8_1X24;
 	chan->format.field = V4L2_FIELD_NONE;
-	chan->format.width = XSCD_MAX_WIDTH;
-	chan->format.height = XSCD_MAX_HEIGHT;
+	chan->format.width = XSCD_DEFAULT_WIDTH;
+	chan->format.height = XSCD_DEFAULT_HEIGHT;
 
 	/* Initialize media pads */
 	num_pads = xscd->memory_based ? 1 : 2;
-- 
2.31.1

